testcontainers/src/core/container.rs

```diff
@@ -222,8 +222,9 @@ where
         self.docker_client.stop(&self.id)
     }
 
-    pub fn start(&self) {
+    pub fn start(&mut self) {
```

Comment:
It's definitely breaking change fix, because public interface has been changed.

In order to avoid this I'd consider interior mutability for `ports` instead

Reply:
However, I think it's not really safe to introduce interior mutability here. It's in general would be not really a safe API.

Thus mutable reference makes sense.

Reply:
As this crate hasn't reached 1.0 yet maybe we should just do the breaking change and wait for the next release (0.16.0) to publish it.